lib/api/handler/pokemon_api.dart

```diff
@@ -30,4 +34,46 @@ class PokemonApi {
           .toList();
     });
   }
+
+  Future<PokemonDetails> getPokemonDetails({
+    required String pokemonId,
+  }) async {
+    final uri = baseUri.replace(path: '${baseUri.path}/pokemon/$pokemonId/');
+
+    return await apiClient.dio.getUri(uri).then((response) {
+      List<dynamic> jsonAbilities = response.data['abilities'];
+      List<dynamic> jsonMoves = response.data['moves'];
+      List<dynamic> jsonStats = response.data['stats'];
+      int pokemonHeight = response.data['height'];
+      int pokemonWeight = response.data['weight'];
+      int pokemonBaseExperience = response.data['base_experience'];
+
+      // Get the list of ability
+      List<PokemonAbility> pokemonAbilityList = jsonAbilities
+          .map<PokemonAbility>((dynamic ability) =>
+              PokemonAbility.fromJson(ability['ability'] as Json))
+          .toList();
+
+      // Get the moves of pokemon
+      List<PokemonMove> pokemonMoveList = jsonMoves
+          .map<PokemonMove>(
+              (dynamic moves) => PokemonMove.fromJson(moves['move'] as Json))
+          .toList();
+
+      // Get the stats of pokemon
+      List<PokemonStats> pokemonStatList = jsonStats
+          .map<PokemonStats>(
+              (dynamic stats) => PokemonStats.fromJson(stats as Json))
+          .toList();
+
+      return PokemonDetails(
```

Comment:
why not maximize the PokemonDetails.fromJson instead to parse everything like the way we did for pokemon list? 🙂

for reference take a look at henry's pr if ever 🙂